mtags/src/main/scala-2/scala/meta/internal/mtags/OnDemandSymbolIndex.scala

```diff
@@ -231,11 +234,33 @@ final class OnDemandSymbolIndex(
     val noExtension = toplevel.value.stripSuffix(".").stripSuffix("#")
     List(
       noExtension + ".scala",
-      noExtension + ".java",
-      "java.base/" + noExtension + ".java" // For Java 11.
+      noExtension + ".java"
     )
   }
 
+  private def modulePaths(toplevel: Symbol): List[String] = {
+    if (Properties.isJavaAtLeast("9")) {
+      val noExtension = toplevel.value.stripSuffix(".").stripSuffix("#")
+      val javaSymbol = noExtension.replace("/", ".")
+      try {
+        sourceJars.loadClass(javaSymbol).toList.flatMap { cls =>
+          // note(@tgodzik) Modules are only available in Java 9+, so we need to ivoke this reflectively
+          val module = cls.getClass().getMethod("getModule").invoke(cls)
+          val moduleName = module.getClass().getMethod("getName").invoke(module)
+
+          List(
+            s"$moduleName/$noExtension.java",
+            s"$moduleName/$noExtension.scala"
+          )
+        }
+      } catch {
+        case NonFatal(_) => Nil
```

Comment:
I would do logging if error happens. It will help investigate if something will start failing.

Reply:
The same as in the other case, this might just create noise for the user.

Reply:
Added onError